app/Jobs/ImportTurboVotePosts.php

```diff
@@ -0,0 +1,193 @@
+    {
+        $this->filepath = $filepath;
+
+        $this->authenticatedUserRole = $authenticatedUserRole;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle(SignupService $signupService, PostService $postService)
+    {
+        // Get the file from storage based on path.
+        $file = Storage::get($this->filepath);
+
+        // Read in the file and get the records.
+        $csv = Reader::createFromString($file);
+        $csv->setHeaderOffset(0);
+        $records = $csv->getRecords();
+
+        foreach ($records as $record) {
+            info('Importing record ' . $record['id'], ['record' => $record]);
+
+            $referralCode = $record['referral-code'];
+
+            if ($referralCode) {
+                $referralCodeValues = $this->parseReferralCode(explode(',', $referralCode));
+
+                if (isset($referralCodeValues['northstar_id']) && isset($referralCodeValues['campaign_id']) && isset($referralCodeValues['campaign_run_id'])) {
+                    // Check if a signup exists already.
+                    $signup = Signup::where([
+                        'northstar_id' => $referralCodeValues['northstar_id'],
+                        'campaign_id' => $referralCodeValues['campaign_id'],
+                        'campaign_run_id' => $referralCodeValues['campaign_run_id'],
+                    ])->first();
+
+                    // If the signup doesn't exist, create one.
+                    if (! $signup) {
+                        $signupData = [
+                            'campaign_id' => $referralCodeValues['campaign_id'],
+                            'campaign_run_id' => $referralCodeValues['campaign_run_id'],
+                            'source' => 'turbovote-import',
+                        ];
+
+                        $signup = $signupService->create($signupData, $referralCodeValues['northstar_id']);
+                    }
+                    // Check if a post already exists.
+                    $post = Post::where([
+                        'signup_id' => $signup->id,
+                        'northstar_id' => $referralCodeValues['northstar_id'],
+                        'campaign_id' => $referralCodeValues['campaign_id'],
+                        'type' => 'voter-reg',
+                    ])->first();
+
+                    if (! $post) {
+                        $tvCreatedAtMonth = strtolower(Carbon::parse($record['created-at'])->format('F-Y'));
+                        $postDetails = $this->extractDetails($record);
+
+                        $postData = [
+                            'campaign_id' => $referralCodeValues['campaign_id'],
+                            'northstar_id' => $referralCodeValues['northstar_id'],
+                            'type' => 'voter-reg',
+                            'action' => $tvCreatedAtMonth . '-turbovote',
+                            'status' => $record['voter-registration-status'],
```

Comment:
@sbsmith86 I might have missed something somewhere but I didn't think that we could directly store status as the TurboVote status because it might be something like `Registered` instead of `Accepted` or `Rejected`.

Reply:
We don't do any validation that would not allow different strings to be stored (as long as they are strings). There is some validation around API requests only being able to accept `accepted`, `rejected`, and `pending` but that will not come into play here since we are adding the post ourselves.

Reply:
@sbsmith86 I thought we were going to decide if it should be `accepted`, `pending`, or `rejected` based on the TurboVote status, and then store the raw status in the `details` column, but since we bypass the validation creating the posts this way we're going to go ahead and store the TurboVote status directly in the `status` column. Am I understanding this right?

Reply:
@katiecrane ahh right right. I thiink we did discuss this but I did not implement any translation of turbovote statuses to our statuses. Let me bring this to the team and try to remember what we should do here.

Reply:
@katiecrane thanks for catching this! I forgot about that discussion (I need to write things down ugh) but I spoke with Jen and I think the best/easiest approach for now is just to store the TV status as is. https://dosomething.slack.com/archives/C0YGXUE01/p1517342932000844

Reply:
Haha I have to write everything down too or it disappears so I feel that! That sounds good!